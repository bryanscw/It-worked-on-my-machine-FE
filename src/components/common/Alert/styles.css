.alert-shown {
    opacity: 1;
    transition: all 250ms linear;
}

.alert-hidden {
    display: none;
    transition: all 250ms linear;
}