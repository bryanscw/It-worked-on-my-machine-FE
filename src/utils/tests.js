import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    return <Provider store={store}>{component}</Provider>;
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return <Router history={history}>{component}</Router>
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    return renderWithRedux(renderWithRouter(component, route), initialState);
}