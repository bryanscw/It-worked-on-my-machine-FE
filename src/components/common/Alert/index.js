import React, { Component } from 'react'
import { connect } from 'react-redux';

import { closeError, selectErrorMessage, selectIsVisible } from '../../../redux/ducks/errors';
import './styles.css';

export class Alert extends Component {
    render() {
        const {
            isVisible,
            errorMessage,
        } = this.props;

        return (
            <div className={`alert alert-danger ${isVisible ? 'alert-shown' : 'alert-hidden'}`} ref={alert => this.alert = alert}>
                <strong>Error:</strong> {errorMessage}
                <button type="button" className="close" onClick={this.props.closeError}>
                    <span>&times;</span>
                </button>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    isVisible: selectIsVisible(state),
    errorMessage: selectErrorMessage(state),
});

const dispatchers = {
    closeError,
};

export default connect(mapStateToProps, dispatchers)(Alert);
