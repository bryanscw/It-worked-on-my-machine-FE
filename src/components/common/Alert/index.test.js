import React from 'react'
import { cleanup, wait, fireEvent } from '@testing-library/react'
import Alert from './index.js';
import { renderWithRedux } from '../../../utils/tests.js';

afterEach(cleanup);

it('should take a snapshot', () => {
    const { asFragment } = renderWithRedux(<Alert />, {
        errorsReducer: {
            isVisible: true,
            errorMessage: 'Error message',
        }
    });

    expect(asFragment()).toMatchSnapshot();
})

it('should show error message if isVisible is true', () => {
    const { getByText } = renderWithRedux(<Alert />, {
        errorsReducer: {
            isVisible: true,
            errorMessage: 'Error message',
        }
    });

    const errorMessage = getByText('Error message');
    expect(errorMessage).toBeVisible;
})

it('should show error message if isVisible is false', () => {
    const { getByText } = renderWithRedux(<Alert />, {
        errorsReducer: {
            isVisible: false,
            errorMessage: 'Error message',
        }
    });

    const errorMessage = getByText('Error message');
    expect(errorMessage).not.toBeVisible;
})

it('should hide the alert the close button is clicked', async () => {
    const { getByText } = renderWithRedux(<Alert />, {
        errorsReducer: {
            isVisible: true,
            errorMessage: 'Error message',
        }
    });

    const errorMessage = getByText('Error message');
    expect(errorMessage).toBeVisible;

    fireEvent.click(getByText('Ã—'));
    await wait(() => expect(errorMessage).not.toBeVisible);
})