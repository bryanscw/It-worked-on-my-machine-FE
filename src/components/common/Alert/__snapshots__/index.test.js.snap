// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger alert-shown"
  >
    <strong>
      Error:
    </strong>
     Error message
    <button
      class="close"
      type="button"
    >
      <span>
        Ã—
      </span>
    </button>
  </div>
</DocumentFragment>
`;
